.gitbook/contents/hc/3_write-smart-contract.md

```diff
@@ -0,0 +1,192 @@
+# Write the Smart Contract
+
+Now we can write the smart contract, which will call our previously created off-chain handler. Our contract has two purposes:
+
+1. Intentionally burn ETH gas (in order to test gas estimation logic).
+2. Increment a counter variable each time the contract is called based on certain inputs and errors.
+
+You can find the needed `HybridAccount` contract along with its dependencies in our [repository](https://github.com/bobanetwork/account-abstraction-hc/contracts/samples/HybridAccount.sol). Additionally, the contract is pulled in as a submodule when you checkout our [`rundler-hc` repository](https://github.com/bobanetwork/rundler-hc/crates/types/contracts/lib/account-abstraction/).
+
+## Waste Gas
+
+To start our contract, we need to import our `HybridAccount` contract and define functions `countFail()` and `justemit()`, the second of which will be used to emit the event `CalledFrom` for troubleshooting:
+
+```solidity
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.12;
+
+import "../samples/HybridAccount.sol";
+
+  function countFail() public pure {
+      revert("count failed");
+  }
+
+  function justemit() public {
+      emit CalledFrom(msg.sender);
+  }
+
+  event CalledFrom(address sender);
+```
+
+To implement our function to waste gas, we can leverage a `for` loop to increase the cost of transacting:
```

Comment:
This is more for representation purposes rather than a requirement.

Reply:
Wasting gas? Or the `for` loop?